GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/PrefetchVerbTests.cs

```diff
@@ -133,11 +133,12 @@ private void PostFetchJobShouldComplete()
 
             // Wait first, to hopefully ensure the background thread has
             // started before we check for the lock file.
+            int timeoutCount = 0;
             do
             {
                 Thread.Sleep(500);
             }
-            while (this.fileSystem.FileExists(postFetchLock));
+            while (this.fileSystem.FileExists(postFetchLock) && ++timeoutCount < 100);

```

Comment:
Same comment here, I don't think this is the hang on Mac because this method is only called by `PrefetchCleansUpStalePrefetchLock` which is not currently running on Mac.  These aren't running as the lock files are not deleted on Mac (by design):

```// TODO(Mac): Handle that lock files are not deleted on Mac, they are simply unlocked```